docs/developer-feature-tutorial.md

```diff
@@ -99,74 +132,104 @@ Now, run zulip terminal and open web app from a different account. Start composi
+                self.controller.view.set_footer_text()
+            else:
+                raise RuntimeError("Unknown typing event operation")
 ```
-If the conditions are satisfied, we display `x is typing` if the `op` is `start` and display help message if the `op` is `stop`.
+The above code uses `set_footer_text()` method implemented the `View` class in `ui.py` which should default to setting a random Help message when not being used.
+
+If the conditions are satisfied, we display `<sender> is typing...` if the `op` is `start` and display help message if the `op` is `stop`.
 
 There are two parts to updating a widget in urwid:
-* Changing the widget
-* Updating the screen
+* Changing the widget, which is done by View
+* Updating the screen, which is done by Controller
 
-This line of code,
-```python
-self._w.footer.set_text([
-                    ' ',
-                    ('code', user['full_name']),
-                    ' is typing...'
-                ])
-```
-changes the footer text, and this
+These are done by the highlighted lines accordingly in `set_footer_text`
 ```python
-self.controller.update_screen()
+@asynch
+def set_footer_text(self, text_list: Optional[List[Any]]=None, duration: Optional[float]=None) -> None:
+    if text_list is None:
+        text = self.get_random_help()
+    else:
+        text = text_list
+>>> self._w.footer.set_text(text)
+>>> self.controller.update_screen()
+    if duration is not None:
+        assert duration > 0
+        time.sleep(duration)
+        self.set_footer_text()
 ```
-updates the screen to display the changes. This fully implements the typing feature.
 
-### Writing tests
+## Writing tests
+
+We use `pytest` to write all our tests. We have to write new tests, modify current ones or delete old ones whenever needed.
 
-Now, we update the tests for `register_initial_desired_events` by adding `typing`
+For this feature, we update the tests for `register_initial_desired_events` by adding `typing`
 to the event types.
-```diff
+```python
 # tests/core/test_core.py
-
-        event_types = [
-            'message',
-            'update_message',
-            'reaction',
-+           'typing',
-        ]
-        controller.client.register.assert_called_once_with(
-                                   event_types=event_types,
-                                   apply_markdown=True)
+  event_types = [
+      'message',
+      'update_message',
+      'reaction',
+      'subscription',
+>>>   'typing',
+      'update_message_flags',
+      'update_display_settings',
+  ]
 ```
 
-`test_handle_typing_event` in `test_ui.py` implements testing for `handle_typing_event`. Please read it to understand how to write tests for a new function in zulip terminal.
+and we create a new function `test_handle_typing_event` in `test_model.py` which implements testing for `handle_typing_event`. 
+
+Please read it to understand how to write tests for a new function in zulip terminal.
```

Comment:
"zulip terminal" and "Zulip Terminal" have been used interchangeably throughout the tutorial. I would suggest keeping it consistent using the capitalized term everywhere.